### PR TITLE
🏏 Add CricketCast Extension

### PR DESCRIPTION
**Pull request by sampoder**
## Description

Allows you to check the cricket scores of ongoing games.

## Type of change

**New extension**

## Screencast

https://user-images.githubusercontent.com/39828164/148332824-128ce014-35aa-4acf-8aea-a3ea365eea6c.mov

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)

### CODE REVIEW

File: extensions/cricketcast/package.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  "name": "cricketcast",
+  "title": "CricketCast",
+  "description": "Check the cricket score through Raycast!",
+  "icon": "command-icon.png",
+  "author": "sampoder",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Show The Cricket Scores",
```
Comment (pernielsentikaer) [inline extensions/cricketcast/package.json:12]:
```suggestion
      "title": "Cricket Scores",
```

File: extensions/cricketcast/src/index.jsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+function Actions({ item }) {
+  return (
+    <ActionPanel title={item.title}>
+      <ActionPanel.Section>{item.link && <OpenInBrowserAction url={item.link} />}</ActionPanel.Section>
+      <ActionPanel.Section>
+        {item.link && (
+          <CopyToClipboardAction content={item.link} title="Copy Link" shortcut={{ modifiers: ["cmd"], key: "." }} />
+        )}
+      </ActionPanel.Section>
+    </ActionPanel>
```
Comment (pernielsentikaer) [inline extensions/cricketcast/src/index.jsx:15]: Better only having one ActionPanel.Section, then will the second item automatically get a modifier

File: extensions/cricketcast/src/index.jsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+export default function Command() {
+  const [state, setState] = useState({});
+
+  useEffect(() => {
+    async function fetchStories() {
+      try {
+        let XMLdata = await fetch("http://static.cricinfo.com/rss/livescores.xml").then((r) => r.text());
+        let JSONdata = JSON.parse(await convert.xml2json(XMLdata, { compact: true, spaces: 4 }));
```
Comment (pernielsentikaer) [inline extensions/cricketcast/src/index.jsx:26]:
```suggestion
        let JSONdata = JSON.parse(convert.xml2json(XMLdata, {compact: true, spaces: 4}));
```
Unnecessary await removed

File: extensions/cricketcast/src/index.jsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        let matches = JSONdata.rss.channel.item.map((match) => ({
+          title: match.title["_text"].replace(" *", "*"),
+          id: match.guid["_text"].replace("http://www.cricinfo.com/ci/engine/match/", "").replace(".html", ""),
```
Comment (pernielsentikaer) [inline extensions/cricketcast/src/index.jsx:29]:
```suggestion
          id: match.guid["_text"].replace( /\D+/g, ''),
```
Could be replaced with this instead, the current format is always the same.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @sampoder 👋 

Thanks for the contribution, I left a few suggestions and comments, could you look into them?

There is also an image in the root on the extension (command-icon.png) seems like it should be removed to?

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by sampoder** [comment]
Thanks @pernielsentikaer, just committed some fixes!

**Review by pernielsentikaer — APPROVED** [review]
Hi @sampoder 👋 

Thanks for the corrections, I can't find anything else that should be corrected 🔥 I'm going to accept it and then will my colleague do a final review before it's finally merged.

**Pull request merged**